src/main/java/org/logstash/dissect/Dissector.java

```diff
@@ -0,0 +1,101 @@
+package org.logstash.dissect;
+
+import com.logstash.Event;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Dissector {
+    private final Pattern pattern = Pattern.compile("(.*?)%\\{(.*?)\\}");
+    private final List<Field> fields = new ArrayList<>();
+    private final List<Field> saveableFields = new ArrayList<>();
+    private final Map<Field, ValueRef> fieldValueRefMap = new HashMap<>();
+    private int initialOffset = 0;
+    private Field lastField;
+
+    public Dissector(String mapping) {
+        if (mapping.isEmpty()) return;
```

Comment:
In the plugin code we validate that the config is a Hash but not that the values are non-empty strings.
I can raise an argument error in the constructor, but I can also, at line 70 in `JavaDissectorLibrary.java` return if the `mapping` RubyString is empty.
It is not really a stop-the-world error if a user supplies an empty string value in the Logstash config while they are testing - for example.